src/EventHub/EventHub/Cmdlets/NetworkRuleSet/SetAzureEventHubNetworkrule.cs

```diff
@@ -0,0 +1,112 @@
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.ResourceManager.Common.Tags;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands.NetworkruleSet
+{
+    /// <summary>
+    /// 'Set-AzEventHubNamespace' Cmdlet updates the specified Eventhub Namespace
+    /// </summary>
+    [Cmdlet("Set", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "EventHubNetworkRuleSet", SupportsShouldProcess = true, DefaultParameterSetName = NetwrokruleSetPropertiesParameterSet), OutputType(typeof(PSNetworkRuleSetAttributes))]
+    public class SetAzureEventHubNetworkrule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 0, HelpMessage = "Resource Group Name.")]
+        [ResourceGroupCompleter]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetInputObjectParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [Parameter(Mandatory = true, ParameterSetName = NetworkRuleSetResourceIdParameterSet, ValueFromPipelineByPropertyName = true, Position = 1, HelpMessage = "EventHub Namespace Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Name { get; set; }       
+
+        [Parameter(Mandatory = false, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true,  HelpMessage = "Default Action for NetwrokeuleSet")]
+        [PSArgumentCompleter("Allow", "Deny")]
+        [PSDefaultValue(Value ="Deny")]
+        public string DefaultAction { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 2, HelpMessage = "List of IPRuleSet")]
+        [ValidateNotNullOrEmpty]
+        public PSNWRuleSetIpRulesAttributes[] IPRule { get; set; }
+
+        [Parameter(Mandatory = true, ParameterSetName = NetwrokruleSetPropertiesParameterSet, ValueFromPipelineByPropertyName = true, Position = 3, HelpMessage = "List of VirtualNetworkRules")]
```

Comment:
@v-Ajnava as mentioned, `-ResourceId` should be the only parameter with `ValueFromPipelineByPropertyName` -- it looks like a lot of other parameters still have this property, so I would double check across your files that this is resolved

Reply:
Resolved